api/v1alpha1/addresspool_types.go

```diff
@@ -20,6 +20,19 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type BgpAdvertisement struct {
+	// The aggregation-length advertisement option lets you “roll up” the /32s into a larger prefix.
+	// +kubebuilder:validation:Minimum=1
+	AggregationLength int `json:"aggregationLength,omitempty" yaml:"aggregation-length,omitempty"`
+
+	// BGP LOCAL_PREF attribute which is used by BGP best path algorithm,
+	// Path with higher localpref is preferred over one with lower localpref.
+	LocalPref uint32 `json:"localpref,omitempty" yaml:"localpref,omitempty"`
```

Comment:
For consistency, should LocalPref be camelCase for JSON and kebab-case for YAML?
Given this is a user API, fixing it later carries with the cost of a depreciation period.

```suggestion
	LocalPref uint32 `json:"localPref,omitempty" yaml:"local-pref,omitempty"`
```

Reply:
I agree for localPref however the yaml need to match the exact configuration in configmap which is `localpref` not `local-pref`